### PR TITLE
Updates to the mp-items-menu and added mp-icon-input

### PR DESCRIPTION
**Pull request by jbwyme**

### CODE REVIEW

File: lib/widgets/items-menu/index.js [hunk 1]
```diff
@@ -74,17 +74,16 @@ class ItemsMenu extends Component {
...
-    super.attachedCallback(...arguments);
-
-    this.handleScroll = ev => {
-      const scrollBottom = ev.target.scrollTop + ev.target.offsetHeight;
-      if (this.state.isProgressive && scrollBottom + PROGRESSIVE_LIST_BUFFER_PX >= ev.target.scrollHeight) {
-        this.increaseProgressiveListSize();
-      }
-    };
-    this.addEventListener(`scroll`, this.handleScroll);
+  set keydownProxy(domElement) {
```
Comment (jbwyme) [inline lib/widgets/items-menu/index.js]: can be used in host jade like:
```
mp-items-menu(props={keydownProxy: $component})
```

Comment (mduan) [inline lib/widgets/items-menu/index.js]: Can't think of a better name for it right now, but what about appending "El"? So `keydownProxyEl`.

Comment (tdumitrescu) [inline lib/widgets/items-menu/index.js]: re: naming, I'd try to make it something that makes it obvious that it's an element, even as simply as `keydownProxyEl`. My suggestions: `keyboardEl`, `keyboardListenerEl`

File: lib/widgets/items-menu/index.js [hunk 2]
```diff
@@ -108,28 +115,46 @@ class ItemsMenu extends Component {
...
+      case `search-filter`:
+        this.update({searchFilter: newVal});
+        break;
+      case `empty-message`:
+        this.update({emptyMessage: newVal});
+        break;
+      case `sections`:
+        this.update({sections: this.getJSONAttribute(`sections`)});
+        break;
+      case `progressive`:
```
Comment (mduan) [inline lib/widgets/items-menu/index.js]: This change is not backwards compatible with existing usages of `mp-items-menu`, so remember to add the progressive attr to callers when you publish this PR.

File: lib/widgets/items-menu/index.js [hunk 3]
```diff
@@ -226,13 +251,17 @@ class ItemsMenu extends Component {
     }
   }
 
-  handleClickItem(item) {
-    this.dispatchEvent(new CustomEvent(`clickItem`, {detail: {item: this.unwrapItem(item)}}));
+  handleClickItem(item, ctrlKey = false) {
+    this.dispatchEvent(new CustomEvent(`clickItem`, {detail: {item: this.unwrapItem(item), ctrlKey}}));
```
Comment (mduan) [inline lib/widgets/items-menu/index.js]: rename `ctrlKey` to `modifierKey`?

**Pull request closed without merging**